pkg/cninetwork/cni_network.go

```diff
@@ -131,43 +147,33 @@ func DeleteCNINetwork(ctx context.Context, cni gocni.CNI, client *containerd.Cli
 	return errors.Wrapf(containerErr, "Unable to find container: %s, error: %s", name, containerErr)
 }
 
-// GetIPAddress returns the IP address of the created container
-func GetIPAddress(result *gocni.CNIResult, task containerd.Task) (net.IP, error) {
-	// Get the IP of the created interface
-	var ip net.IP
-	for ifName, config := range result.Interfaces {
-		if config.Sandbox == netNamespace(task) {
-			for _, ipConfig := range config.IPConfigs {
-				if ifName != "lo" && ipConfig.IP.To4() != nil {
-					ip = ipConfig.IP
-				}
-			}
-		}
-	}
-	if ip == nil {
-		return nil, fmt.Errorf("unable to get IP address for: %s", task.ID())
-	}
-	return ip, nil
-}
+// GetIPAddress returns the IP address from container based on name and PID
+func GetIPAddress(name string, PID uint32) (string, error) {
+	processName := fmt.Sprintf("%s-%d", name, PID)
+	CNIDir := path.Join(CNIDataDir, defaultNetworkName)
 
-func GetIPfromPID(pid int) (*net.IP, error) {
-	// https://github.com/weaveworks/weave/blob/master/net/netdev.go
-
-	peerIDs, err := ConnectedToBridgeVethPeerIds(defaultBridgeName)
+	files, err := ioutil.ReadDir(CNIDir)
 	if err != nil {
-		return nil, fmt.Errorf("unable to find peers on: %s %s", defaultBridgeName, err)
-	}
-
-	addrs, addrsErr := GetNetDevsByVethPeerIds(pid, peerIDs)
-	if addrsErr != nil {
-		return nil, fmt.Errorf("unable to find address for veth pair using: %v %s", peerIDs, addrsErr)
+		return "", fmt.Errorf("failed to read CNI dir for container %s: %v", name, err)
 	}
 
-	if len(addrs) > 0 && len(addrs[0].CIDRs) > 0 {
-		return &addrs[0].CIDRs[0].IP, nil
+	for _, file := range files {
+		f, err := os.Open(filepath.Join(CNIDir, file.Name()))
+		if err != nil {
+			return "", fmt.Errorf("failed to open CNI IP file for %s/%s: %v", CNIDir, file.Name(), err)
+		}
+		reader := bufio.NewReader(f)
```

Comment:
@carlosedp do you remember what this is doing?

I've added output of what the files look like in the PR text.

Reply:
I remember each IP allocation saves the container IP on a file in this dir...
Need to take a look and if needed rework this whole PR.

Reply:
Look at the PR description if you like. I provided two samples for you to view without setting anything up.

Just curious about the approach chosen for parsing a JSON file.

Reply:
It turns out this was a text file. I've documented it in a follow-up commit and have added a unit test for it. cc @LucasRoesler